src/qr_login.rs

```diff
@@ -0,0 +1,141 @@
+//! Types for QR code login
+
+use matrix_sdk_crypto::types::qr_login;
+use url::Url;
+use wasm_bindgen::prelude::*;
+
+use crate::vodozemac::Curve25519PublicKey;
+
+/// The mode of the QR code login.
+///
+/// The QR code login mechanism supports both, the new device, as well as the
+/// existing device to display the QR code.
+///
+/// The different modes have an explicit one-byte identifier which gets added to
+/// the QR code data.
+#[wasm_bindgen]
+#[derive(Debug)]
+pub enum QrCodeMode {
+    /// The new device is displaying the QR code.
+    Login,
+    /// The existing device is displaying the QR code.
+    Reciprocate,
+}
+
+impl From<qr_login::QrCodeMode> for QrCodeMode {
+    fn from(value: qr_login::QrCodeMode) -> Self {
+        match value {
+            qr_login::QrCodeMode::Login => Self::Login,
+            qr_login::QrCodeMode::Reciprocate => Self::Reciprocate,
+        }
+    }
+}
+
+/// Data for the QR code login mechanism.
+///
+/// The [`QrCodeData`] can be serialized and encoded as a QR code or it can be
+/// decoded from a QR code.
+#[wasm_bindgen]
+#[derive(Debug)]
+pub struct QrCodeData {
+    inner: qr_login::QrCodeData,
+}
+
+#[wasm_bindgen]
+impl QrCodeData {
+    /// Create new [`QrCodeData`] from a given public key, a rendezvous URL and,
+    /// optionally, a homeserver URL.
+    ///
+    /// If a homeserver URL is given, then the [`QrCodeData`] mode will be
+    /// [`QrCodeMode::Reciprocate`], i.e. the QR code will contain data for the
+    /// existing device to display the QR code.
+    ///
+    /// If no homeserver is given, the [`QrCodeData`] mode will be
+    /// [`QrCodeMode::Login`], i.e. the QR code will contain data for the
+    /// new device to display the QR code.
```

Comment:
please bear in mind that `` [`QrCodeData`] `` doesn't work in the documentation generated by `wasm_bindgen`: see https://matrix-org.github.io/matrix-rust-sdk-crypto-wasm/classes/QrCodeData.html#constructor.

You need to use tsdoc-style `{@link ... }` links.

Reply:
I changed the style in: https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/125/commits/2c9bd6fa0f62ef9e39e90c08292be2a7b643971c.